### PR TITLE
[doc] `Getopt::Long::Parser`: Improvements


### PR DESCRIPTION
**Pull request by rwp0**
- Clarify Synopsis
- Format code
- Fix incorrect syntax (`new`)
- Scope variables
- Fix few spelling

### CODE REVIEW

**Comment by rwp0** [comment]
@sciurius please tell if I need to modify something to make it mergeable

**Comment by rwp0** [comment]
@sciurius Any chance of reviewing this?

**Comment by sciurius** [comment]
Most changes are okay. Please remove the C<...> in the HEAD line. This line is used by many external tools and I'm not sure all of them will understand this.

**Comment by rwp0** [comment]
> Most changes are okay. Please remove the C<...> in the HEAD line. This line is used by many external tools and I'm not sure all of them will understand this.

I guess you mean the NAME line.

Okay, I will amend that.

**Comment by rwp0** [comment]
@sciurius `C<>` removed, commit amended and force-pushed

File: lib/Getopt/Long/Parser.pm [hunk 1]
```diff
@@ -13,28 +13,28 @@ our $VERSION = 2.57;
 
 =head1 NAME
 
-Getopt::Long::Parser - Getopt::Long object oriented interface
+Getopt::Long::Parser - Getopt::Long object-oriented interface
```
Comment (rwp0) [inline lib/Getopt/Long/Parser.pm:16]: Removed `C<>` from `Getopt::Long` amending my commit

**Pull request merged**